sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java

```diff
@@ -159,9 +157,16 @@ public double getRequestCharge() {
         return Double.parseDouble(value);
     }
 
-    StoreResponse setCosmosDiagnostics(CosmosDiagnostics cosmosDiagnostics) {
-        this.cosmosDiagnostics = cosmosDiagnostics;
-        return this;
+    /**
+     * Static factory method to create serializable store response to be used in CosmosDiagnostics
+     * @param storeResponse store response
+     * @return serializable store response
+     */
+    public static StoreResponse createSerializableStoreResponse(StoreResponse storeResponse) {
```

Comment:
For this one I would prefer having different types - either via composition - putting all the info required for diagnostics into StoreResponseStatistics or StoreResponseDiagnostics class that is a field in StoreResponse (peer to content) and then just return that subclass for diagnostics. Or inheritence - (content on sub class, diagnostics relevant fields on base class). My preference is composition. But I think it would be good to isolate them. Same probably could be done for CosmosException to make it easier to understand the model.

Reply:
Agree on the composition model for `StoreResponse` and `CosmosException`. Easier to map the diagnostics as well. However, I have few concerns: 
1. Doing this change for AEMO, and changing too many things in one hotfix. 
2. StoreResponse and CosmosException are natively used in a lot of other diagnostics places (like Tracer, and CosmosPagedFlux, etc) - testing all of them puts a lot of extra effort, which might affect the timeline. 

In my opinion, we should re-model the diagnostics in a separate iteration, and release it after some good testing. 

Also, if we want to put that effort now, it is okay to do so, but worth discussing this in regards to AEMO timeline.